packages/test/test-end-to-end-tests/src/testSnapshotCache.ts

```diff
@@ -16,16 +16,19 @@ export interface ValueWithSnapshot {
 }
 
 export class TestSnapshotCache implements IPersistedCache {
-	private readonly cache = new Map<string, ValueWithSnapshot>();
+	private readonly cache = new Map<string, unknown>();
 	private readonly versionCache = new Map<string, ValueWithSnapshot>();
 	private readonly versionToCacheKey = new Map<string, string>();
-	public async get(entry: ICacheEntry): Promise<ValueWithSnapshot | undefined> {
+	public async get(entry: ICacheEntry): Promise<any | undefined> {
 		const key = getKeyForCacheEntry(entry);
 		return this.cache.get(key);
 	}
-	public async put(entry: ICacheEntry, value: ValueWithSnapshot): Promise<void> {
+	public async put(entry: ICacheEntry, value: any): Promise<void> {
 		const key = getKeyForCacheEntry(entry);
 		this.cache.set(key, value);
+		if (value.value?.snapshotTree?.id === undefined) {
```

Comment:
What is the logic behind this?

Reply:
The logic is to check to see if we are storing a snapshot or not. If it's not a snapshot, then this will be undefined, and thus we will not store it in the snapshot cache.